main/EDTypesMod.F90

```diff
@@ -14,15 +14,16 @@ module EDTypesMod
   use FatesLitterMod,        only : litter_type
   use FatesLitterMod,        only : ncwd
   use FatesConstantsMod,     only : n_anthro_disturbance_categories
-
+  use FatesConstantsMod,     only : days_per_year
+  
   implicit none
   private               ! By default everything is private
   save
 
   integer, parameter, public :: maxPatchesPerSite  = 14   ! maximum number of patches to live on a site
   integer, parameter, public :: maxPatchesPerSite_by_disttype(n_anthro_disturbance_categories)  = &
                                                      (/ 10, 4 /)  !!! MUST SUM TO maxPatchesPerSite !!!
-  integer, parameter, public :: maxCohortsPerPatch = 100  ! maximum number of cohorts per patch
+  integer, parameter, public :: maxCohortsPerPatch = 300  ! maximum number of cohorts per patch
```

Comment:
can we revert this to original number if no age tracking is happening?

Reply:
Yes, but somewhere it will need to be increased if cohort age tracking is on.

Reply:
ok, seems like there is the potential for circular dependencies here.  maybe the simplest thing to do is, rather than using the slightly cryptic `if (ED_val_cohort_age_fusion_tol < 10000.0_r8)` condition a few times in the code as it is now, it would make more sense to have some logical variable called like `use_cohort_age_tracking`, which you set as true or false during initialization (possibly using the same assessment of ED_val_cohort_age_fusion_tol), and then refer to that here and elsewhere.  but where is the best place to put such a flag to avoid circular dependencies?  @rgknox do you know?

Reply:
For the time being, perhaps EDParamsMod.F90?

Reply:
Having a spot for these types of flags is part of the wishlist for refactoring FatesInterfaceMod.F90.  That refactor will create a new module that has just datastructures, including the various control flags, but would not have many of the setup procedures, so it would have little dependency issues.

Reply:
I changed maxCohortsPerPatch back to 100 and made a local variable maxCohortsPerPatch_age_tracking in EDCohortDynamicsMod.F90 that is 300 and that overwrites maxCohortsPerPatch if age tracking is on. Not sure if this is ok or if it should be in EDTypesMod.F90 also? I got rid of the cryptic ED_val_cohort_age_fusion_tol switch and made a cohort_age_tracking switch.